tools-local/tasks/mobile.tasks/AndroidAppBuilder/ApkBuilder.cs

```diff
@@ -244,30 +240,58 @@ public class ApkBuilder
+
+        return (alignedApk, packageId);
+    }
+
+    private void SignApk(string apkPath, string apksigner)
+    {
+        string defaultKey = Path.Combine(OutputDir, "debug.keystore");
+        string signingKey = string.IsNullOrEmpty(KeyStorePath) ?
+            defaultKey : Path.Combine(KeyStorePath, "debug.keystore");
 
-        string signingKey = Path.Combine(OutputDir, "debug.keystore");
-        if (!string.IsNullOrEmpty(KeyStorePath))
-            signingKey = Path.Combine(KeyStorePath, "debug.keystore");
         if (!File.Exists(signingKey))
         {
-            Utils.RunProcess(keytool, "-genkey -v -keystore debug.keystore -storepass android -alias " +
+            Utils.RunProcess("keytool", "-genkey -v -keystore debug.keystore -storepass android -alias " +
                 "androiddebugkey -keypass android -keyalg RSA -keysize 2048 -noprompt " +
                 "-dname \"CN=Android Debug,O=Android,C=US\"", workingDir: OutputDir, silent: true);
         }
-        else
+        else if (Path.GetFullPath(signingKey) != Path.GetFullPath(defaultKey))
         {
             File.Copy(signingKey, Path.Combine(OutputDir, "debug.keystore"));
         }
+        Utils.RunProcess(apksigner, $"sign --min-sdk-version {MinApiLevel} --ks debug.keystore " +
+            $"--ks-pass pass:android --key-pass pass:android {apkPath}", workingDir: OutputDir);
+    }
+
+    public void ReplaceFileInApk(string file)
+    {
+        if (string.IsNullOrEmpty(AndroidSdk))
+            AndroidSdk = Environment.GetEnvironmentVariable("ANDROID_SDK_ROOT");
 
-        // 6. Sign APK
+        if (string.IsNullOrEmpty(AndroidSdk) || !Directory.Exists(AndroidSdk))
+            throw new ArgumentException($"Android SDK='{AndroidSdk}' was not found or incorrect (can be set via ANDROID_SDK_ROOT envvar).");
 
-        Utils.RunProcess(apksigner, $"sign --min-sdk-version {MinApiLevel} --ks debug.keystore " +
-            $"--ks-pass pass:android --key-pass pass:android {alignedApk}", workingDir: OutputDir);
+        if (string.IsNullOrEmpty(BuildToolsVersion))
+            BuildToolsVersion = GetLatestBuildTools(AndroidSdk);
 
-        Utils.LogInfo($"\nAPK size: {(new FileInfo(alignedApk).Length / 1000_000.0):0.#} Mb.\n");
+        if (string.IsNullOrEmpty(MinApiLevel))
+            MinApiLevel = DefaultMinApiLevel;
 
-        return (alignedApk, packageId);
+        string buildToolsFolder = Path.Combine(AndroidSdk, "build-tools", BuildToolsVersion);
+        string aapt = Path.Combine(buildToolsFolder, "aapt");
+        string apksigner = Path.Combine(buildToolsFolder, "apksigner");
+
+        string apkPath = Directory.GetFiles(Path.Combine(OutputDir, "bin"), "*.apk").First();
+        Utils.RunProcess(aapt, $"remove -v bin/{Path.GetFileName(apkPath)} {file}", workingDir: OutputDir);
+        Utils.RunProcess(aapt, $"add -v bin/{Path.GetFileName(apkPath)} {file}", workingDir: OutputDir);
```

Comment:
In order to replace `lib/x86_64/libSystem.Native.so`, it looks like there would need to be a file `libSystem.Native.so` nested in `lib/x86_64` in the projects directory. 
Would the use case call for only generating files of the same name to replace the pre-existing one?
If it would be an iterative process where previously generated files would want to kept (`libSystem.Native.so_one` `libSystem.Native.so_two`), would it be worth having some sort of `targetFile` `replacementFile` pair?

Reply:
it depends on aapt behavior - it asks you to put files in a dir the way it will be copied to apk.

so the user has to update any file in the apk folder he wants and run the task.

Reply:
Right, I'm still not very well acquainted with why/when a user would replace a specific file in the apk, but I was just thinking of a way for more convenience if the user had multiple different versions of the file they wanted to try to substitute in.

Reply:
it's needed when you e.g. test changes in libSystem.Native.so and don't want to wait the full apk rebuild. So you just copy it to the final apk dir where needed and run re-sync. I don't think it makes any sense to make a copy of the old file - it's just a way to improve build iterations, especially with native code (I need it for my Android Crypto work)